pkg/lib/serverTime.js

```diff
@@ -258,42 +267,21 @@ export function ServerTime() {
         return status;
     };
 
-    const custom_ntp_config_file = cockpit.file("/etc/systemd/timesyncd.conf.d/50-cockpit.conf",
-                                                { superuser: "try" });
+    function get_timesync_backend() {
+        return python.spawn(get_timesync_backend_py, [], { superuser: "try", err: "message" })
+                .then(data => data.trim());
+    }
+
+    function get_custom_ntp_timesyncd() {
+        const custom_ntp_config_file = cockpit.file("/etc/systemd/timesyncd.conf.d/50-cockpit.conf",
+                                                    { superuser: "try" });
 
-    self.get_custom_ntp = function () {
-        /* We only support editing the configuration of
-         * systemd-timesyncd, by dropping a file into
-         * /etc/systemd/timesyncd.conf.d.  We assume that timesyncd is
-         * used when:
-         *
-         * - systemd-timedated is answering for
-         *   org.freedesktop.timedate1 as opposed to, say, timedatex.
-         *
-         * - systemd-timesyncd is enabled (false if chrony is being used)
-         *
-         * The better alternative would be to have an API in
-         * o.fd.timedate1 for managing the list of NTP server
-         * candidates.
-         */
         const result = {
-            supported: false,
+            backend: "timesyncd",
             enabled: false,
             servers: []
         };
 
-        if (!timedate1_service.exists || timedate1_service.unit.Id !== "systemd-timedated.service") {
-            console.log("systemd-timedated not in use, ntp server configuration not supported");
-            return Promise.resolve(result);
-        }
-
-        if (!timesyncd_service.enabled) {
-            console.log("systemd-timesyncd not enabled, ntp server configuration not supported");
```

Comment:
I don't see this check anywhere any more. unit_exists() in the Python helper does *not* check for "enabled".

Reply:
This was a wrong test. systemd-timesyncd.service is not enabled when NTP is off.